core/services/chainlink/application.go

```diff
@@ -192,13 +197,79 @@ type ApplicationOpts struct {
 	NewOracleFactoryFn         standardcapabilities.NewOracleFactoryFn
 }
 
+type ApplicationHeartbeat struct {
+	commonservices.Service
+	eng *commonservices.Engine
+
+	beat time.Duration
+	lggr logger.Logger
+}
+
+func NewApplicationHeartbeat(lggr logger.Logger) ApplicationHeartbeat {
+	h := ApplicationHeartbeat{
+		beat: ApplicationHeartbeatSeconds * time.Second,
+		lggr: lggr,
+	}
+	h.Service, h.eng = commonservices.Config{
+		Name:  "NodeHeartbeat",
+		Start: h.start,
+		Close: h.close,
+	}.NewServiceEngine(lggr)
+	return h
+}
+
+func (h *ApplicationHeartbeat) start(_ context.Context) error {
+	// Setup beholder resources
+	gauge, err := beholder.GetMeter().Int64Gauge("heartbeat")
+	if err != nil {
+		return err
+	}
+	count, err := beholder.GetMeter().Int64Gauge("heartbeat_count")
+	if err != nil {
+		return err
+	}
+
+	cme := custmsg.NewLabeler()
+
+	// Define tick functions
+	beatFn := func(engCtx context.Context) {
```

Comment:
IMHO alternate names for context confuse things and lead to accidental misuse.
```suggestion
	beatFn := func(ctx context.Context) {
```
If there are multiple contexts in scope leading to a linter problem, then it is typically cleaner to separate the nested scopes. In this case, a method could work. Although I don't see a nesting problem.

Reply:
>IMHO alternate names for context confuse things and lead to accidental misuse.

Oh, huh. I liked that it was clearer for me. Even after putting this down for a while I immediately knew when i picked it back up what deadline considerations were in scope for the method. 

I do see that it could mess with tooling. It also is clearer in this case because the context passed to `start` is unused.